tests/testsuite/docscrape.rs

```diff
@@ -48,6 +48,127 @@ fn basic() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2021"
+                authors = []
+
+                [dependencies]
+                a = { path = "crates/a" }
+                b = { path = "crates/b" }
+            "#,
+        )
+        .file("src/lib.rs", "pub use a::*;\npub use b::*;")
+        // Examples
+        .file(
+            "examples/one.rs",
+            r#"use foo::*;
+fn main() {
+    let foo = Foo::new("yes".into());
+    foo.maybe();
+}"#,
+        )
+        .file(
+            "examples/two.rs",
+            r#"use foo::*;
+fn main() {
+    let foo: Foo = Foo::new("yes".into());
+    let bar = Bar { amount: 1200. };
+    println!("{}", bar.pa());
+}"#,
+        )
+        // `a` crate
+        .file(
+            "crates/a/Cargo.toml",
+            r#"
+                [package]
+                name = "a"
+                version = "0.0.1"
+                authors = []
+        "#,
+        )
+        .file(
+            "crates/a/src/lib.rs",
+            r#"
+#[derive(Debug)]
+pub struct Foo {
+    foo: String,
+    yes: bool,
+}
+
+impl Foo {
+    pub fn new(foo: String) -> Self {
+        Self { foo, yes: true }
+    }
+
+    pub fn maybe(&self) {
+        if self.yes {
+            println!("{}", self.foo)
+        }
+    }
+}"#,
+        )
+        // `b` crate
+        .file(
+            "crates/b/Cargo.toml",
+            r#"
+                [package]
+                name = "b"
+                version = "0.0.1"
+                authors = []
+        "#,
+        )
+        .file(
+            "crates/b/src/lib.rs",
+            r#"
+#[derive(Debug)]
+pub struct Bar {
+    /// pressure in Pascal
+    pub amount: f32,
+}
+
+impl Bar {
+    pub fn pa(&self) -> f32 {
+        self.amount
+    }
+}"#,
+        )
+        .build();
+
+    p.cargo("doc -Zunstable-options -Zrustdoc-scrape-examples --no-deps")
+        .masquerade_as_nightly_cargo(&["rustdoc-scrape-examples"])
+        .with_stderr_unordered(
+            "\
+[CHECKING] a v0.0.1 ([CWD]/crates/a)
+[CHECKING] b v0.0.1 ([CWD]/crates/b)
+[CHECKING] foo v0.0.1 ([CWD])
+[SCRAPING] foo v0.0.1 ([CWD])
+[DOCUMENTING] foo v0.0.1 ([CWD])
+[FINISHED] [..]
+[GENERATED] [CWD]/target/doc/foo/index.html",
+        )
+        .run();
+
+    let doc_html = p.read_file("target/doc/foo/struct.Foo.html");
+    assert!(doc_html.contains("Examples found in repository"));
+    // `Foo` is used in the two examples so since there are more than one example, we should have
+    // this sentence.
+    assert!(doc_html.contains("More examples"));
+    let doc_html = p.read_file("target/doc/foo/struct.Bar.html");
+    assert!(doc_html.contains("Examples found in repository"));
+    // There is only only usage of `Bar` so we shouldn't have this sentence in the docs.
+    assert!(!doc_html.contains("More examples"));
+}
```

Comment:
As we just need to make sure examples are found, can we simplify this?
- Only one dependency
- Only one example

I'm most interested in minimizing the number of assumptions we make in this test on rustdoc's output so its less likely to fail on upgrade and its easier to update when it does fail.